Modules/_datetimemodule.c

```diff
@@ -6608,19 +6631,23 @@ _datetime_exec(PyObject *module)
     PyDateTime_Epoch = new_datetime(1970, 1, 1, 0, 0, 0, 0,
                                     PyDateTime_TimeZone_UTC, 0);
     if (PyDateTime_Epoch == NULL) {
+        PyMem_Free(capi);
         return -1;
     }
 
     /* module initialization */
     if (PyModule_AddIntMacro(module, MINYEAR) < 0) {
+        PyMem_Free(capi);
         return -1;
     }
     if (PyModule_AddIntMacro(module, MAXYEAR) < 0) {
+        PyMem_Free(capi);
         return -1;
     }
 
-    x = PyCapsule_New(&CAPI, PyDateTime_CAPSULE_NAME, NULL);
+    x = PyCapsule_New(capi, PyDateTime_CAPSULE_NAME, datetime_destructor);
```

Comment:
Call get_datetime_capi() here and remove all previous calls to PyMem_Free(capi).

Reply:
done.